### PR TITLE
PCHR-778: Add task design

### PR DESCRIPTION
**Pull request by kkalamarski**
## 1. Problem
Modal was too wide.
width_before

### Solution
Modal width was changed from `600px` to `500px` to meet the design.
width_after

## 2. Problem
Due date should be set to today's date by default (pre-filled).
date_before

### Solution
Due date is now set to today's date.
date_after

## 3. Problem
When a task is created, Target and Assignee fields are empty in the dashboard, unless you refresh the page.

**Reason:**
method `updateCache()` was altering cache with object what was lacking `display_name` field. Therefore template was trying to display field that no longer exist in model. 
missing_name_before

### Solution
`display_name` field was added to updateCache method. 
missing_name_after

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_forms.scss [hunk 1]
```diff
@@ -5,6 +5,19 @@ form {
     padding-right: 0;
   }
 
+  /* Bootstrap override */
+  .input-group-addon {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_forms.scss]: What's the reason for moving this outside `.form-horizontal`?

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-task_test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    'tasks-assignments/app'
+], function (angular, moment) {
+    'use strict';
+
+    describe('ModalTaskCtrl', function () {
+        var ctrl, modalInstance, scope, HR_settings;
+
+        beforeEach(module('civitasks.appDashboard'));
+        beforeEach(inject(function ($controller, $rootScope) {
+            modalInstance = {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-task_test.js]: Separate this into its own function, so you can have a more concise `beforeEach` implementation:
```js
ctrl = $controller('ModalTaskCtrl', {
    $scope: scope,
    $modalInstance: fakeModalInstance(), // <---
    data: {},
    HR_settings: HR_settings
 });
```

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-task_test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                close: jasmine.createSpy('modalInstance.close'),
+                dismiss: jasmine.createSpy('modalInstance.dismiss'),
+                result: {
+                    then: jasmine.createSpy('modalInstance.result.then')
+                }
+            };
+
+            scope = $rootScope.$new();
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-task_test.js]: Use `$scope`, to mirror the variable name used by angular

File: uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-task_test.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+            ctrl = $controller('ModalTaskCtrl', {
+                $scope: scope,
+                $modalInstance: modalInstance,
+                data: {},
+                HR_settings: HR_settings
+            });
+        }));
+
+        describe('Task should have a default (pre-filled) date.', function () {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-task_test.js]: Add a test for when the task already has a date

File: uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_forms.scss (hunk 1 above)
Comment (kkalamarski) [inline uk.co.compucorp.civicrm.tasksassignments/scss/civitasks/partials/_forms.scss]: Scope of this style shouldn't be limited only to `.form-horizontal` but all forms inside Tasks&Assignments.

**Comment by AkA84** [comment]
Looks good to me, but just to be sure can you check (if you haven't done it already) that the modal in "edit mode" still looks ok?

**Comment by kkalamarski** [comment]
I did check. Looks ok.

**Comment by AkA84** [comment]
Approved

**Pull request merged**